### PR TITLE
[BLO-905] Remove signer from multisig

### PR DESCRIPTION
**Pull request by diegodelrieu**
### Issue / feature description

`Remove signer from multisig`

### Changes

- Add actions to delete signer (one at a time for now)
- Add delete button in screen

### CODE REVIEW

File: packages/extension/src/background/transactions/transactionMessaging.ts [hunk 1]
```diff
@@ -499,6 +499,44 @@ export const handleTransactionMessage: HandleMessage<
...
+
+        const signersPayload = {
+          entrypoint: "removeSigners",
+          calldata: stark.compileCalldata({
+            new_threshold: newThreshold.toString(),
+            signers_to_remove_len: "1",
+            signers_to_remove: utils.hexlify(
+              utils.base58.decode(signerToRemove),
+            ),
+          }),
```
Comment (dhruvkelawala) [inline packages/extension/src/background/transactions/transactionMessaging.ts:515]:
```suggestion
          calldata: stark.compileCalldata({
            new_threshold: newThreshold.toString(),
            signers_to_remove: utils.hexlify(
              utils.base58.decode(signerToRemove),
            ),
          }),
```

Don't need to add `*_len`

File: packages/extension/src/shared/call/changeMultisigSignersCall.ts [hunk 2]
```diff
@@ -17,3 +17,21 @@ export const isAddMultisigSignersCall = (
...
+export interface RemoveMultisigSignersCall extends Call {
+  entrypoint: "removeSigners"
+}
+
+export const isRemoveMultisigSignersCall = (
+  call: Call,
+): call is AddMultisigSignersCall => {
+  try {
+    if (call.contractAddress && call.entrypoint === "removeSigners") {
+      validateAndParseAddress(call.contractAddress)
```
Comment (dhruvkelawala) [inline packages/extension/src/shared/call/changeMultisigSignersCall.ts:30]:
```suggestion
      validateAndParseAddress(call.contractAddress)
```
This seems better https://github.com/argentlabs/argent-x/blob/bb5d62371cedfa90f6876f207985b74a1d9515c2/packages/extension/src/ui/services/addresses.ts#L76

File: packages/extension/src/ui/features/multisig/MultisigOwnersScreen.tsx [hunk 3]
```diff
@@ -58,15 +68,25 @@ const MultisigOwners = ({ account }: { account: Account }) => {
...
+                  justifyContent="space-between"
+                  alignItems="center"
                 >
                   <H6 color="white">{formatTruncatedSignerKey(signer)}</H6>
-                </Box>
+                  <Button
+                    borderRadius="full"
+                    backgroundColor="neutrals.900"
+                    onClick={() => handleRemoveOwnerClick(signer)}
+                    px="1em"
```
Comment (dhruvkelawala) [inline packages/extension/src/ui/features/multisig/MultisigOwnersScreen.tsx:85]: Why using `em` units? I think it's better to just using `px={1}` or `px={0.5}`

File: packages/extension/src/shared/wallet.model.ts [hunk 4]
```diff
@@ -56,6 +56,12 @@ export type AddOwnerMultisiPayload = {
   currentThreshold?: number
 }
 
+export type RemoveOwnerMultisigPayload = {
+  address: string
+  newThreshold: number
+  signerToRemove: string
+}
+
 export type UpdateMultisigThresholdPayload = {
```
Comment (dhruvkelawala) [inline packages/extension/src/shared/wallet.model.ts:65]: I don't understand why these types are inside `wallet.model.ts`? They shouldn't be here imo

**Review by dhruvkelawala — CHANGES_REQUESTED** [review]

File: packages/extension/src/shared/wallet.model.ts (hunk 4 above)
Comment (diegodelrieu) [reply packages/extension/src/shared/wallet.model.ts:65]: true good catch

File: packages/extension/src/shared/call/changeMultisigSignersCall.ts (hunk 2 above)
Comment (diegodelrieu) [reply packages/extension/src/shared/call/changeMultisigSignersCall.ts:30]: I saw this in the 2fa calls files, I'm indifferent about either but I think we should only have one way of checking this

**Review by simonheys — APPROVED** [review]

**Review by dhruvkelawala — APPROVED** [review]

**Pull request merged**